app/org/sagebionetworks/bridge/play/controllers/ScheduledActivityController.java

```diff
@@ -94,13 +111,31 @@ public Result updateScheduledActivities() throws Exception {
         }
         ClientInfo clientInfo = getClientInfoFromUserAgentHeader();
         
+        DateTime accountCreatedOn = session.getUser().getAccountCreatedOn();
+        if (accountCreatedOn == null) {
+            Study study = studyService.getStudy(session.getStudyIdentifier());
+            User user = session.getUser();
+            // Everyone should have an ID at this point... otherwise sessions are hanging out for over a week.
+            if (user.getId() == null) {
+                accountCreatedOn = DateTime.now();
+                LOG.debug("neither accountCreatedOn nor ID exist in session, using current time");
```

Comment:
Minor: Sessions can live on forever (see https://sagebionetworks.jira.com/browse/BRIDGE-1024), so not everyone may have the user ID in their session. Worst case scenario, we'll need to flush everyone's sessions.